R/Flood_CamML_App/app.R

```diff
@@ -288,85 +288,548 @@ ui <- material_page(
+  
+  # Function to apply to each
+  render_cam_image <- function(cam_name, alt_name){
+    out_image <- renderImage({
+      outfile <- str_c(cam_name, ".jpg")
+      list(src = outfile,
+           alt = alt_name,
+           width = "100%"#, height="180px"
+      )
+    }, deleteFile=F)
+    
+    return(out_image)
+  }
+  
+  # Run Each Camera
+  output$mirlo_picture <- render_cam_image(cam_name = "Mirlo", alt_name = "Mirlo Beach")
+  output$northdock_picture <- render_cam_image(cam_name = "Northdock", alt_name = "NorthDock")
+  output$southdock_picture <- render_cam_image(cam_name = "SouthDock", alt_name = "SouthDock")
+  output$southocracoke_picture <- render_cam_image(cam_name = "SouthOcracoke", alt_name = "SouthOcracoke")
+  
+  
+  
+  
+  
+  #--------------- Camera Feedback UI ----------------------
+  
+  # 2. Display for image box / model classification
+  
+  # Function to apply to each
+  # takes the camera name, the reactive time, and the model predictions
+  render_camera_ui <- function(cam_name, cam_time, model_prediction, id_suffix = ""){
+    
+    # string prep for naming patterns for UI elements
+    # option to add suffix for "_unsupervised" ui elements
+    name_lcase <- tolower(cam_name)
+    img_output_id <- str_c(name_lcase, "_picture", id_suffix)
+    radio_button_id <- str_c(name_lcase, "_button_select", id_suffix)
+    button_clear <- str_c(name_lcase, "_clear", id_suffix)
+    
+    # UI generation for radio buttons
+    camera_button_ui <- renderUI({
+      w$show()
+      box(width="100%",
+          # height=300,
+          solidHeader = T,
+          status = "success", #ifelse(model_prediction >= 0.6, "danger", ifelse(model_prediction <0.6 & model_prediction >=0.4, "warning", "success"))
+          title  = cam_name,
+          align  = "center",
+          # Display Cam Image
+          imageOutput(img_output_id,
+                      height="100%"),
+          # Datetime for image
+          p(paste0(cam_time %>% lubridate::with_tz("America/New_York"), " EDT/EST")),
+          # Inline boxes for user feedback
+          div(style="display:inline-block",
+              shinyWidgets::radioGroupButtons(inputId = radio_button_id, 
+                                              choiceNames = c("Flooding", "Not Sure", "No Flooding"), 
+                                              choiceValues = c("Flooding", "Not Sure", "No Flooding"), 
+                                              justified = F, 
+                                              selected = character(0), 
+                                              checkIcon = list(yes = icon("ok",lib="glyphicon")))
+          ),
+          
+          # clear selection button
+          div(style="display:inline-block",
+              actionButton(inputId = button_clear, 
+                           label = "Clear", 
+                           class = "btn btn-primary", 
+                           style = "font-size:10pt;color:white")
+          )
+      )
```

Comment:
Function to display box around each camera image displaying model output, also sets up the ID's and UI for the feedback buttons